tests/unit/test_cli.py

```diff
@@ -21,3 +27,56 @@ def test_cli_main(self, mock_exporter):
         config = Config(port=mock_port, level=mock_level, enable_collectors=mock_enable_collectors)
         main(config)
         mock_exporter.assert_called_once_with(mock_port)
+
+    @pytest.mark.parametrize(
+        "enable_collectors",
+        [
+            (["collector.mega_raid"]),
+            (["collector.ipmi_dcmi", "collector.ipmi_sel", "collector.ipmi_sensor"]),
+            (["collector.lsi_sas_2", "collector.lsi_sas_3"]),
+            (["collector.mega_raid", "collector.poweredge_raid"]),
+        ],
+    )
+    @patch.object(__main__, "Exporter")
+    def test_cli_main_enable_collector(self, mock_exporter, enable_collectors):
+        config = Config(port=10000, level="INFO", enable_collectors=enable_collectors)
+
+        collector_registries = get_collector_registries(config)
+
+        for collector in enable_collectors:
+            if collector in collector_registries:
+                collector_registries[collector] = Mock()
+
+        with patch.object(__main__, "get_collector_registries", return_value=collector_registries):
+            main(config)
+
+        for collector in enable_collectors:
+            mock_exporter.return_value.register.assert_any_call(
+                collector_registries.get(collector)
+            )
+
+        for name, collector in collector_registries.items():
+            if name.lower() not in enable_collectors:
+                with pytest.raises(AssertionError):
+                    mock_exporter.return_value.register.assert_called_with(collector)
+
+
+def test_get_collector_registries():
+    """Assert all the keys."""
+    config = Config(port=10000, level="INFO", enable_collectors=[])
+
+    # Every time someone add/remove collector should change this.
+    collector_registries = get_collector_registries(config)
+    assert len(collector_registries) == 9
+
+    assert list(collector_registries.keys()) == [
```

Comment:
1. If we write the unit test more flexibile. We only need to check with `set`. We don't need the function return in a specific order (`list`). After we check the `set` is matched, we don't need to check the length. it's duplicated.

2. We can check if the value is not None.